js/tooltip.js

```diff
@@ -139,6 +144,12 @@
       $(obj.currentTarget).data('bs.' + this.type, self)
     }
 
+    if (obj instanceof $.Event) {
+      self.inState[obj.type == 'focusout' ? 'focus' : 'hover'] = false
+    }
+
+    if (self.inState.click || self.inState.focus || self.inState.hover) return
```

Comment:
or better, create a method like: "self.isInStateTrue()" which returns a boolean since you use that below as well